mlir/lib/Dialect/SCF/TransformOps/SCFTransformOps.cpp

```diff
@@ -511,7 +532,8 @@ transform::LoopFuseSibling::apply(transform::TransformRewriter &rewriter,
     return diag;
 
   // Check if the target can be fused into source.
-  if (!isForallWithIdenticalConfiguration(target, source)) {
+  if (!isForallWithIdenticalConfiguration(target, source) &&
+      !isForWithIdenticalConfiguration(target, source)) {
```

Comment:
So this check will `dyn_cast` source and target to one kind, do some checking, and forget about casted results. `fuseSiblings` below will do `dyn_cast` _again_. Could we avoid that by having a positive check for `isForallWith...` and a direct call to `fuseIndependent...` in the body of the conditional?

Reply:
Agreed that this was a bit wasteful. Have now adopted your suggestion (and gotten rid of the `fuseSiblings` function).